FrontEnd/src/components/Navigation.js

```diff
@@ -8,71 +8,75 @@ import { Row, Col } from 'react-materialize';
 import componentCSS from './Navigation.scss';
 
 class Navigation extends React.Component {
-	constructor(props) {
-		super(props);
+    constructor(props) {
+        super(props);
+        this.state = {
 
-		this.state = {
-			pages: [
-				{
-					icon: 'book',
-					text: 'Library',
-					ref: '/index.html',
-					class: 'valign-wrapper',
-				},
-				{
-					icon: 'settings',
-					text: 'Settings',
-					ref: '/settings.html',
-					class: 'valign-wrapper',
-				},
-				{
-					icon: 'face',
-					text: 'Admin',
-					ref: '/admin.html',
-					class: 'valign-wrapper',
-				},
-			],
-		};
+            pages: [
+                {
+                    icon: 'book',
+                    text: 'Library',
+                    class: 'valign-wrapper active',
+					isSelected: true,
+					callbackFn: () => {this.switchSelected(this.state.pages[0].text);},
+                },
+                {
+                    icon: 'settings',
+                    text: 'Settings',
+                    class: 'valign-wrapper',
+					isSelected: false,
+					callbackFn: () => {this.switchSelected(this.state.pages[1].text);},
+                },
+                {
+                    icon: 'face',
+                    text: 'Admin',
+                    class: 'valign-wrapper',
+					isSelected: false,
+					callbackFn: () => {this.switchSelected(this.state.pages[2].text);},
+                },
+            ],
 
-		switch (this.props.isActive) {
-			case 'Library':
-				this.state.pages[0].class += ' active';
-				break;
+        };
+    }
 
-			case 'Settings':
-				this.state.pages[1].class += ' active';
-				break;
+    render() {
+        return (
+            <div className={'navigation'}>
+                <Row>
+                    <Col s={12}>
+                        <ul className="mt0 mb0">
+                            {this.state.pages.map((navList) => (
+                                <li key={uuidv4()}>
+                                    <div className={navList.class} onClick={navList.callbackFn}>
+                                        <span className="material-icons md-18">{navList.icon}</span>
+                                        <span className="link-text">{navList.text}</span>
+                                    </div>
+                                </li>
+                            ))}
+                        </ul>
+                    </Col>
+                </Row>
+            </div>
+        );
+    }
+    
+    switchSelected(id) {
```

Comment:
you need to bind all "internal functions"

Reply:
this this is done